### PR TITLE
Add search in rotated array solution

### PR DESCRIPTION
**Pull request by fbokovikov**

### CODE REVIEW

File: src/main/kotlin/leetcode/challenge/common/TreeUtils.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package leetcode.challenge.common;
+
+import java.util.*;
+
+public class TreeUtils {
+
+    public static List<Integer> bfs(TreeNode root) {
+        List<Integer> result = new ArrayList<>();
```
Comment (fbokovikov) [inline src/main/kotlin/leetcode/challenge/common/TreeUtils.java:8]: It's required for tests

File: src/main/kotlin/leetcode/challenge/week3/RestoreBinaryTree.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package leetcode.challenge.week3;
+
+import leetcode.challenge.common.TreeNode;
+
+public class RestoreBinaryTree {
```
Comment (fbokovikov) [inline src/main/kotlin/leetcode/challenge/week3/RestoreBinaryTree.java:5]: Submitted from the first attempt!!!

File: src/main/kotlin/leetcode/challenge/week3/RestoreBinaryTree.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    public TreeNode bstFromPreorder(int[] preorder) {
+        return buildSubtree(preorder, new int[] { 0 }, Integer.MAX_VALUE, Integer.MIN_VALUE);
```
Comment (DmitryBokovikov) [inline src/main/kotlin/leetcode/challenge/week3/RestoreBinaryTree.java:7]: no-no-no, just pass 0

File: src/main/kotlin/leetcode/challenge/week3/RestoreBinaryTree.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    }
+
+    private TreeNode buildSubtree(int[] preorder, int[] curNodeIndex, int maxValue, int minValue) {
```
Comment (DmitryBokovikov) [inline src/main/kotlin/leetcode/challenge/week3/RestoreBinaryTree.java:10]: private TreeNode buildSubtree(int[] preorder, int curNodeIndex, int maxValue, int minValue) {

File: src/main/kotlin/leetcode/challenge/week3/RestoreBinaryTree.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+        int curIndex = curNodeIndex[0];
```
Comment (DmitryBokovikov) [inline src/main/kotlin/leetcode/challenge/week3/RestoreBinaryTree.java:11]: don't need this variable

File: src/main/kotlin/leetcode/challenge/week3/RestoreBinaryTree.java [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+        if (curIndex == preorder.length || preorder[curIndex] > maxValue || preorder[curIndex] < minValue) {
+            return null;
+        }
```
Comment (DmitryBokovikov) [inline src/main/kotlin/leetcode/challenge/week3/RestoreBinaryTree.java:14]: ++curNodeIndex;

File: src/main/kotlin/leetcode/challenge/week3/RestoreBinaryTree.java (hunk 4 above)
Comment (DmitryBokovikov) [inline src/main/kotlin/leetcode/challenge/week3/RestoreBinaryTree.java:10]: it's not curNodeIndex - misleading name of vaiable

**Pull request merged**